pkgs/development/python-modules/bsddb3/default.nix

```
@@ -21,12 +22,9 @@ buildPythonPackage rec {
   build-system = [ setuptools ];
 
   buildInputs = [ pkgs.db ];
-
-  doCheck = pythonOlder "3.12"; # distutils usage
-
-  checkPhase = ''
-    ${python.interpreter} test.py
-  '';
+  # See : https://github.com/NixOS/nixpkgs/pull/311198#discussion_r1599257522
+  # More details here : https://www.jcea.es/programacion/pybsddb.htm
+  broken = pythonAtLeast "3.10";
```

```suggestion
  disabled = pythonAtLeast "3.10";
```

`broken` doesn't seem to do anything...
If you use `disabled` it will not even try to build it.

---

Or do you want to mark it as broken, but still let people use it if they allow broken packages?

---

Yes I believe it would be better to give a choice, only allowing broken package can be done by defining `NIXPKGS_ALLOW_BROKEN=1`

---

We use disabled for those cases where we know the package will not work.